llvm/lib/Target/AIE/AIEClusterBaseAddress.cpp

```diff
@@ -136,242 +117,152 @@ class AIEClusterBaseAddress : public MachineFunctionPass {
-  }
-
-  // Process all the encountered load and stores in the basic block.
-  // We find the base register of the Ld/St MI and using that base register, we
-  // find following ptr adds which have the same base register. If we find such
-  // a ptr add, we create an entry in the ChainedPtrAdds map.
-  void processLoadOrStore(MachineInstr &MI, MachineRegisterInfo &MRI,
-                          std::map<MachineInstr *, PtrAdd> &ChainedPtrAdds) {
-    Register LdOrStPtrReg = MI.getOperand(1).getReg();
-    // Find the base defining reg of the given MI and the ptr offset.
-    auto BaseRegAndOffset = findBaseReg(MRI, LdOrStPtrReg);
-    if (!BaseRegAndOffset.has_value())
-      return;
-    if ((!EnableChainsForScalarLdSt && getLoadStoreSize(MI) <= 32) ||
-        (!EnableChainsForVectorLdSt && getLoadStoreSize(MI) >= 256))
-      return;
-    Register BaseReg = BaseRegAndOffset->first;
-    // If the base reg is used in any of the successive MBBs, then we don't want
-    // to chain the corresponding ptr adds. Since that would introduce a COPY
-    // and increase reg pressure.
-    if (isRegUsedInSuccessiveMBBs(MI.getParent(), BaseReg, MRI))
-      return;
-    int64_t BasePtrOffset = BaseRegAndOffset->second;
-    // Find the next G_PTR_ADD MachineInstr that comes after the given
-    // MachineInstr and has the same base register.
-    MachineInstr *NextPtrAddMI = findNextPtrAddForReg(MI, MRI, BaseReg);
-    if (!NextPtrAddMI)
-      return;
-    auto [OffsetReg, ConstVal] = getPtrAddOffsetInfo(*NextPtrAddMI, MRI);
-    if (!ConstVal)
-      return;
-    int64_t NextPtrOffset = ConstVal->Value.getSExtValue();
-    PtrAdd PA;
-    PA.PtrAddMI = NextPtrAddMI;
-    PA.BaseReg = BaseReg;
-    PA.NewOffset = NextPtrOffset - BasePtrOffset;
-    ChainedPtrAdds[&MI] = PA;
-  }
-
-  unsigned getLoadStoreSize(const MachineInstr &MI) {
-    return (*MI.memoperands_begin())->getSizeInBits().getValue();
+    return Changed;
   }
 
-  // Get a set of all reachable MBBs from a given MBB.
-  // Loops are handled using the ReachableMBBs set, once we encounter any
-  // reachable MBB from a particular MBB, we store it in the set and continue if
-  // we find it again. Lastly, we remove the current MBB from the set in case it
-  // comes up in the successive basic blocks.
-  std::set<MachineBasicBlock *> findReachableMBBs(MachineBasicBlock *MBB) {
-    std::set<MachineBasicBlock *> ReachableMBBs;
-    SmallVector<MachineBasicBlock *, 8> Worklist;
-    Worklist.append(MBB->succ_begin(), MBB->succ_end());
-    while (!Worklist.empty()) {
-      MachineBasicBlock *CurrMBB = Worklist.pop_back_val();
-      if (!ReachableMBBs.insert(CurrMBB).second)
-        continue;
-      Worklist.append(CurrMBB->succ_begin(), CurrMBB->succ_end());
+  // Get all candidates, i.e. groups of G_PTR_ADDs in the same
+  // basic block that shares the same input pointer.
+  void getChainingCandidates(MachineBasicBlock &MBB,
+                             RegToPtrAddMap &RegPtrUseMap) {
```

Comment:
How about changing this to `RegToPtrAddMap getChainingCandidates(MachineBasicBlock &MBB)`? This makes it clearer what comes out of this function?